counsel.el

```diff
@@ -6268,12 +6271,24 @@ The properties include:
 This variable is suitable for addition to
 `savehist-additional-variables'.")
 
+(defvar counsel-compile-history-exact-src nil
```

Comment:
Should this be a `defvar-local`?

Reply:
Yes, and renamed:

(defvar-local counsel-compile-history-match-buffer-file-name nil
  "Make `counsel-compile-get-filtered-history' match buffer name.
This is usually set to nil except for the special case of
standalone files where matching what is usually $HOME can cause
confusion.")

Reply:
Note that "buffer name" and `buffer-file-name` are not the same thing, and if used interchangeably could lead to confusion.